.github/workflows/docker-build.yml

```diff
@@ -0,0 +1,48 @@
+name: Docker-Build Pipeline
+
+on:
+  push:
+    branches:
+      - main
+  # TODO: disable until caching system is implemented
```

Comment:
I think this can be saved for a separate PR in the future. We need to first figure out a caching strategy to prevent rebuilding dockers that we've already built before, unless the docker tag has changed.